File: crates/domain-scan-core/tests/fixtures/go/schemas.go
package main

type UserDTO struct {
	ID    string `json:"id"`
	Name  string `json:"name"`
	Email string `json:"email,omitempty"`
	Age   *int   `json:"age,omitempty"`
}

type CreateUserRequest struct {
	Name  string `json:"name" db:"name"`
	Email string `json:"email" db:"email"`
}

type InternalConfig struct {
	Host string
	Port int
}

File: crates/domain-scan-core/tests/fixtures/swift/schemas.swift
import Foundation

// Codable struct
struct UserDTO: Codable {
    let id: String
    let name: String
    let email: String
    let age: Int
}

// Codable struct with optional fields
struct ProductResponse: Codable, Equatable {
    let id: String
    let title: String
    let description: String?
    let price: Double
}

// Codable class
class OrderModel: Codable {
    var orderId: String
    var items: [String]
    var total: Double
}

// Non-codable struct (should NOT be detected as schema)
struct InternalConfig {
    let maxRetries: Int
    let timeout: Double
}
